common/fs/fs.go

```diff
@@ -0,0 +1,52 @@
+package fs
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"syscall"
+)
+
+// T is the filesystem interface type.
+type T interface {
+	ReadDir(string) ([]os.FileInfo, error)
+	ReadFile(string) ([]byte, error)
+	Lstat(string, *syscall.Stat_t) error
+	Stat(string, *syscall.Stat_t) error
+	Open(string) (io.ReadWriteCloser, error)
+}
+
+type realFS struct{}
+
+// FS is the way you should access the filesystem.
+var FS T = realFS{}
```

Comment:
Maybe move this (and `type realFS ...`) below the mocking stuff so it's all together.